src/main/java/org/sqlite/core/NativeDB.c

```diff
@@ -1777,7 +1777,7 @@ static void freeCommitHandlerCtx(JNIEnv *env, void *ctx) {
 
 void clear_commit_listener(JNIEnv *env, jobject nativeDB, sqlite3 *db) {
     sqlite3_commit_hook(db, NULL, NULL);
-    sqlite3_update_hook(db, NULL, NULL);
+    sqlite3_rollback_hook(db, NULL, NULL);
```

Comment:
I think you found and fixed the problem. Nice work (:
Because we're closing the db, it may have an empty transaction that it rollsback before closing if autocommit is disabled.
Since the rollback hook was still present due to this error, it tries to call the rollback function, which is set to null due to the next line.
Then it crashes.

Reply:
unfortunately the tests don't pass, while it works on my machine, and i don't know why

Reply:
The recompiled binaries were added to the incorrect directories in git, so the CI is using the old binaries.

On your local machine (and mine), when you ran make, it put the new binary in the correct directory.

Reply:
Example: Windows/x86/sqlitejdbc.dll should be src/main/resources/org/sqlite/native/Windows/x86/sqlitejdbc.dll
etc.

Reply:
oh, let me have a look at that. I thought this worked well, but i may have botched it 🤦🏻